tck/features/SetAcceptance.feature

```diff
@@ -0,0 +1,332 @@
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the node using a simple expression
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET (n).name = 'neo4j'
+      RETURN n
+      """
+    Then the result should be:
+      | n                    |
+      | (:A {name: 'neo4j'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the node using an expression
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET (CASE WHEN true THEN n END).name = 'neo4j'
+      RETURN n
+      """
+    Then the result should be:
+      | n                    |
+      | (:A {name: 'neo4j'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the relationship using a simple expression
+    Given any graph
+    And having executed:
+      """
+      CREATE ()-[:REL]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r:REL]->()
+      SET (r).name = 'neo4j'
+      RETURN r
+      """
+    Then the result should be:
+      | r                      |
+      | [:REL {name: 'neo4j'}] |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the relationship using an expression
+    Given any graph
+    And having executed:
+      """
+      CREATE ()-[:REL]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r:REL]->()
+      SET (CASE WHEN true THEN r END).name = 'neo4j'
+      RETURN r
+      """
+    Then the result should be:
+      | r                      |
+      | [:REL {name: 'neo4j'}] |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Setting a property to null removes the property
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A {name: 'Michael', age: 35})
+      """
+    When executing query:
+      """
+      MATCH (n)
+      WHERE n.name = 'Michael'
+      SET n.name = null
+      RETURN n
+      """
+    Then the result should be:
+      | n              |
+      | (:A {age: 35}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Add a label to a node
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET n:FOO
+      RETURN n
+      """
+    Then the result should be:
+      | n        |
+      | (:A:FOO) |
+    And the side effects should be:
+      | +labels | 1 |
+
+  Scenario: Adding a list property
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET n.x = [1, 2, 3]
+      RETURN extract(i IN n.x | i/2.0) AS x
```

Comment:
Also, no `extract()`. Use a list comprehension, like so:

`RETURN [i IN n.x | i / 2.0]`